src/plugins/workspace/public/components/workspace_column/workspace_colum.test.tsx

```diff
@@ -21,27 +21,77 @@ describe('workspace column in saved objects page', () => {
   ];
   coreSetup.workspaces.workspaceList$.next(workspaceList);
 
-  it('should show workspace name correctly', () => {
+  it('should show workspace name badge correctly', () => {
     const workspaces = ['ws-1', 'ws-2'];
-    const { container } = render(<WorkspaceColumn coreSetup={coreSetup} workspaces={workspaces} />);
+    const { getByTestId, getByText, container } = render(
+      <WorkspaceColumn coreSetup={coreSetup} workspaces={workspaces} />
+    );
+    const badge = getByTestId('workspace-column-more-workspaces-badge');
+    expect(badge).toBeInTheDocument();
+    fireEvent.click(badge);
+    expect(getByTestId('workspace-column-popover')).toBeInTheDocument();
+    expect(getByText('foo')).toBeInTheDocument();
+    expect(getByText('bar')).toBeInTheDocument();
```

Comment:
Nit: we could use findByText or findByTestId for such kind of assertion, those commands have retry mechanism out of the box.